### PR TITLE
Fix text locale encoding issues on windows

### PR DESCRIPTION
**Pull request by rvl**
Relates to #703.

# Overview

- Forces the current windows console into UTF-8 mode.
- UTF-8 is far from standard on windows. But it shouldn't be assumed on any platform.
- This fixes strange characters appearing on windows and test failures with exceptions related to encoding.

### CODE REVIEW

File: lib/cli/src/Cardano/CLI.hs [hunk 1]
```diff
@@ -375,9 +379,9 @@ cmdWalletCreate = command "create" $ info (helper <*> cmd) $ mempty
             getLine prompt parser
         wSndFactor <- do
             let prompt =
-                    "(Enter a blank line if you do not wish to use a second \
-                    \factor.)\n\
-                    \Please enter a 9–12 word mnemonic second factor: "
+                    "(Enter a blank line if you do not wish to use a second " <>
+                    "factor.)\n" <>
+                    "Please enter a 9–12 word mnemonic second factor: "
```
Comment (jonathanknowles) [inline lib/cli/src/Cardano/CLI.hs:372]: Just out of curiosity, what's the reason for changing the line continuation style here?

File: lib/cli/src/Cardano/CLI.hs [hunk 2]
```diff
@@ -1230,13 +1234,26 @@ withLogging configFile minSeverity action = bracket before after (action . snd)
...
+-- other settings.
+--
+-- On Windows the current console code page is changed to UTF-8.
 setUtf8Encoding :: IO ()
+#if WINDOWS
 setUtf8Encoding = do
-    setLocaleEncoding utf8
-    hSetEncoding stdout utf8
-    hSetEncoding stderr utf8
+    let utf8CP = 65001 -- code page number for UTF-8
```
Comment (jonathanknowles) [inline lib/cli/src/Cardano/CLI.hs]: The comment can probably be made redundant if we rename the identifier to something like:

```utf8CodePage```.

File: lib/cli/src/Cardano/CLI.hs (hunk 1 above)
Comment (rvl) [reply lib/cli/src/Cardano/CLI.hs:372]: The `CPP` language extension.

File: lib/cli/src/Cardano/CLI.hs (hunk 2 above)
Comment (rvl) [reply lib/cli/src/Cardano/CLI.hs]: Sure

**Comment by rvl** [comment]
I need to move `setUtf8Encoding` to the `cardano-wallet-core` package to avoid a package dependency cycle.

File: lib/cli/src/Cardano/CLI.hs (hunk 1 above)
Comment (jonathanknowles) [reply lib/cli/src/Cardano/CLI.hs:372]: > The `CPP` language extension.

Understood! At least our use of the CCP extension is limited to just a single file.

**Comment by rvl** [comment]
Rebased onto master.

**Review by KtorZ — APPROVED** [review]

**Pull request merged**